tcf_website/management/commands/load_semester.py

```diff
@@ -252,29 +252,49 @@ def load_section(
             units,
             section_type):
 
+        unique_params = {}
         params = {}
-        fields = {
+
+        unique_fields = {
             'sis_section_number',
             'semester',
+        }
+        fields = {
             'course',
             'topic',
             'units',
-            'section_type'}
+            'section_type'
+        }
+
         for k, v in locals().items():
+            if k in unique_fields and not pd.isnull(v):
+                unique_params[k] = v
             if k in fields and not pd.isnull(v):
                 params[k] = v
 
         # print(locals())
-        section, created = Section.objects.get_or_create(
-            **params
-        )
+        # section, created = Section.objects.update_or_create(
+        #     sis_section_number=unique_params['sis_section_number'],
+        #     semester=unique_params['semester'],
+        #     **params)
+        section = None
+        created = False
+        try:
+            section = Section.objects.get(**unique_params)
+            for key, value in params.items():
+                setattr(section, key, value)
+            section.save()
+        except Section.DoesNotExist:
+            params.update(unique_params)
```

Comment:
Maybe add `# NOTE: Change to params |= unique_params when we switch to Python 3.9` above this line, similar to https://github.com/thecourseforum/theCourseForum2/blob/68ef6be00238a7d737210a7900afc3ddac88f571/tcf_website/views/profile.py#L69